devel/ci/bodhi-ci

```diff
@@ -281,6 +279,44 @@ def integration(archive, concurrency, container_runtime, no_build, failfast, ini
                                archive_path=archive_path))
 
 
+class ProgressReporter(object):
+    """Report progress on the provided jobs.
+
+    Attributes:
+        jobs: A list of the Jobs to report on.
+    """
+
+    def __init__(self, jobs: typing.List['Job']):
+        """
+        Args:
+            jobs: A list of the Jobs to report on.
+        """
+        self._jobs = []
+        for job in jobs:
+            self.register_job(job)
+
+    def register_job(self, job):
+        """Register a job to be reported.
+
+        Args:
+            job (Job): A job to report the status of.
+        """
+        self._jobs.append(job)
+        loop = asyncio.get_event_loop()
+        loop.create_task(self._print_on_complete(job))
+
+    async def _print_on_complete(self, job):
+        """Print the status of a job when it's complete."""
+        await job.complete.wait()
+        self.print_status()
```

Comment:
Oh that's a cool way to do this. I can think of a slight problem though, which is that you don't know when this will happen. Since there are many tasks all waiting for control and there's a limit on how many concurrent tasks can run (defaults to number of processors), the status report won't necessarily print when a job is finished (because some other test task might start instead).

What do you think? It's probably not a big deal since it's just a status report, but I do find the status report to be helpful.

Reply:
Hmm, I'm not sure that could happen, because the call to `self.complete.set()` is done outside of the `async with concurrency_semaphore:` block, so it should not wait for control.
But I may be mistaken there, I haven't played with asyncio much (but I did play a lot with Twisted, or rather Twisted played me a lot).

Reply:
Ah yes, that's a good point. I think this assuages my concern. Thanks!